Scripts/Source/minai_Sex.psc

```diff
@@ -194,84 +194,97 @@ Function StartSexOrSwitchToGroup(actor[] actors, actor akSpeaker, string tags=""
   Main.Info("Sex: Starting/switching for " + actors.Length + " actors (tags: " + tags + ")")
   bool bSpeakerInScene = False
   bool bPlayerInScene = False
-  int i =0
-  int numMales = 0
-  int numFemales = 0
   bool bCanAnimate = True
-  Actor[] actorsInScene
-  sslThreadController Controller
   bool isNewScene = False
   if bHasOstim && minai_UseOStim.GetValue() == 1.0
+    Actor[] ostimActors = new actor[5]
+    ostimActors = actors
     ; Ostim
+    ostimActors = actors
+    tags = ConvertTagsOstim(tags)
     int ActiveOstimThreadID = OActor.GetSceneID(akSpeaker)
     if ActiveOstimThreadID < 0
-      actorsInScene = actors
-      isNewScene = true
-    else
-      actorsInScene = OThread.GetActors(ActiveOstimThreadID)
+      ; akSpeaker is not in an OStim thread
+      if ostimActors.Length == 1
+        OThread.Quickstart(akSpeaker)
+      else
+        ActiveOstimThreadID = OActor.GetSceneID(ostimActors[1])
+        if ActiveOstimThreadID < 0
+          ; Target 1 is not in an OStim thread
+          if OActor.VerifyActors(ostimActors)
+          StartSexScene(ostimActors, bPlayerInScene, tags)
+          EndIf
+        else
+          ; Target 1 is already in an OStim thread
+          ostimActors = OThread.GetActors(ActiveOstimThreadID)
+          ; add akSpeaker to OStim actor array
+          ostimActors = OActorUtil.ToArray(akSpeaker, ostimActors[0],ostimActors[1],ostimActors[2],ostimActors[3])
+          OThread.Stop(ActiveOstimThreadID)
```

Comment:
Having to stop and then restart the thread to add actors to it seems a bit rough. Is there a more seamless way like in sexlab?

Reply:
Unfortunately, not according to OStim author.